media/js/base/translation-bar.es6.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+(function () {
+    const barElem = document.getElementById('translation-bar');
+    if (!barElem) {
+        return false;
+    }
+
+    // Normalize language in the form of ab or ab-CD.
+    function normalize(lang) {
+        return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+            return p1.toLowerCase() + (p2 ? '-' + p2.toUpperCase() : '');
+        });
+    }
+
+    // Returns boolean if lang matches target lang.
+    function isLangMatch(targetLang, lang) {
+        return (
+            targetLang === lang ||
+            targetLang === lang.split('-')[0] ||
+            targetLang.split('-')[0] === lang.split('-')[0]
+        );
+    }
+
+    // Return best matched available language, or `false` if none found.
+    function getBestMatch(acceptLangs, availableLangs) {
+        // Note: Do not use `forEach` here since you can't break the loop and
+        // will return the least best language.
+        for (const acceptLang of acceptLangs) {
+            for (const availableLang of availableLangs) {
+                if (isLangMatch(acceptLang, availableLang)) {
+                    return availableLang;
+                }
+            }
+        }
+        return false;
+    }
+
+    const pageLang = document.documentElement.lang;
+    const availableLangs = Array.from(
+        document.querySelectorAll('link[hreflang]'),
+        (link) => normalize(link.hreflang)
+    ).filter((e) => e !== 'x-DEFAULT');
+    const acceptLangs = navigator.languages.map((lang) => normalize(lang));
+    const bestLang = getBestMatch(acceptLangs, availableLangs);
+    const bestLink = document.querySelector(
+        `link[hreflang='${bestLang}']`
+    ).href;
+
+    let showBar = true;
+    if (
+        !bestLang ||
+        isLangMatch(pageLang, availableLangs[0]) ||
+        isLangMatch(pageLang, bestLang)
+    ) {
+        showBar = false;
+    }
+
+    function createBar(strings, link) {
+        /*
+         Build DOM of the translate bar.
+
+            <aside class="mzp-c-notification-bar">
+                <button class="mzp-c-notification-bar-button" type="button">Close notification</button>
+                <p><a href="{link}">View this page in your language</a></p>
+            </aside>
```

Comment:
Question: I guess this might need some testing once it's ready, but I'm not 100% sure if the design of notification component is going to make it the easiest approach for something that can appear on any page of the site. There's a chance it might overlap existing content / notifications on other pages. Keeping this as something that will only appear above a page's regular content might be a less obtrusive approach perhaps?